pallets/settlement/src/lib.rs

```diff
@@ -1127,15 +1008,14 @@ impl<T: Trait> Module<T> {
                         let status = Self::instruction_leg_status(instruction_id, leg_id);
                         status == LegStatus::ExecutionPending
                     }) {
-                        let result = <Asset<T>>::base_transfer(
+                        if <Asset<T>>::base_transfer(
                             leg_details.from,
                             leg_details.to,
                             &leg_details.asset,
                             leg_details.amount,
-                        );
-                        if let Ok(post_info) = result {
-                            transaction_weight += post_info.actual_weight.unwrap_or_default();
-                        } else {
+                        )
+                        .is_err()
+                        {
                             return Err(leg_id);
                         }
```

Comment:
Can't we just use the `?` operator?

Reply:
I think not because we have to return whenever the error occurs.

Reply:
`?` does return.